File: src/net/t3kt/tctrl/schema/params/BoolParamSchema.java
package net.t3kt.tctrl.schema.params;

import com.google.protobuf.Value;
import java.util.Optional;
import net.t3kt.tctrl.schema.TctrlSchemaProto.ParamSpec;

public final class BoolParamSchema extends ScalarParamSchema<Boolean> {
    BoolParamSchema(ParamSpec spec) {
        super(spec);
    }

    public String getOffHelp() {
        return spec.getOffHelp();
    }

    public String getButtonText() {
        return spec.getButtonText();
    }

    public String getButtonOffText() {
        return spec.getButtonOffText();
    }

    @Override
    protected Optional<Boolean> convert(Value value) {
        return Values.toBool(value);
    }
}

File: win/TctrlCore/Schema/Params/BoolParamSchema.cs
using System;
using Google.Protobuf.WellKnownTypes;
using Tctrl.Schema;

namespace Tctrl.Core.Schema.Params {
    public sealed class BoolParamSchema : ScalarParamSchema<bool> {
        public BoolParamSchema(ParamSpec spec) : base(spec) { }

        protected override bool? Convert(Value value) => value.AsBool();

        public String OffHelp => Spec.OffHelp;
        public String ButtonText => Spec.ButtonText;
        public String ButtonOffText => Spec.ButtonOffText;

    }
}